src/test/java/com/google/api/codegen/metacode/SampleInitCodeTest.java

```diff
@@ -120,58 +136,55 @@ public void testNestedListField() throws Exception {
     InitCodeNode innerList = InitCodeNode.create("0");
     InitCodeNode outerList =
         InitCodeNode.createWithChildren("0", InitCodeLineType.ListInitLine, innerList);
-    InitCodeNode expectedStructure =
+    InitCodeNode outerStructure =
         InitCodeNode.createWithChildren("mylist", InitCodeLineType.ListInitLine, outerList);
+    InitCodeNode expectedStructure = InitCodeNode.newRoot();
+    expectedStructure.mergeChild(outerStructure);
 
-    InitCodeNode actualStructure = FieldStructureParser.parse(fieldSpec);
-    Truth.assertThat(checkEquals(actualStructure, expectedStructure)).isTrue();
+    InitCodeNode actualStructure = InitCodeNode.newRoot();
+    FieldStructureParser.parse(actualStructure, fieldSpec);
+
+    assertNodeEqual(actualStructure, expectedStructure);
   }
 
   @Test
   public void testFormattedField() throws Exception {
     String fieldSpec = "name%entity";
 
-    HashMap<String, InitValueConfig> initValueMap = new HashMap<>();
     InitValueConfig initValueConfig = InitValueConfig.create("test-api", null);
-    initValueMap.put("name", initValueConfig);
+    InitCodeNode outerStructure = InitCodeNode.createWithValue("name", initValueConfig);
```

Comment:
"entity" does not show up anywhere because it has no `=`? Could we add comments to that effect here and in the code?

Reply:
I honestly don't know why this is happen actually, but the test passing implies we're correctly keeping the old behavior.

Reply:
Argh. We should understand the mysterious code, but no need to block this PR on this point